onadata/apps/logger/management/commands/update_attachment_storage_bytes.py

```diff
@@ -0,0 +1,177 @@
+
+                # if `last_xform` is not none, it means that the `user_id` is
+                # different from the previous one in the loop so the user
+                # profile must be updated
+                if last_xform:
+                    self.update_user_profile(
+                        last_xform, total_per_user[last_xform['user_id']]
+                    )
+
+            # write out xform progress
+            if self.verbosity >= 1:
+                self.stdout.write(
+                    f"Calculating attachments for xform_id #{xform['pk']}"
+                    f" (user {xform['user__username']})"
+                )
+            # aggregate total media file size for all media per xform
+            # We cannot get the sum of all attachments for on xform because
+            # we need to make two joins with Instance and XForm models.
+            # It is really slow on big databases. So, to avoid that, we make
+            # another extra query to get all the instance ids to pass it to
+            # Attachment queryset.
+            instance_ids = Instance.objects.values_list('pk', flat=True).filter(
+                xform_id=xform['pk']
+            )
+
+            # It does not seem to have a real limit of number of parameters in
+            # "IN" clause in PostgreSQL but for safety, we use chunks to pass
+            # to `instance_id__in` just in case a form has a huge numbers
+            # of submissions.
+            # See https://stackoverflow.com/questions/1009706/postgresql-max-number-of-parameters-in-in-clause
+            instance_ids_count = len(instance_ids)
+            max_ids_per_query = 5000
+            chunks = [
+                instance_ids[x:x + max_ids_per_query]
+                for x in range(0, instance_ids_count, max_ids_per_query)
+            ]
+            form_attachments_total = 0
+            for idx, chunk in enumerate(chunks):
+                if self.verbosity > 1:
+                    self.stdout.write(
+                        f'\tCalculating total: {idx+1}/{len(chunks)} instance ID'
+                        f' chunks'
+                    )
+                form_attachments = Attachment.objects.filter(
+                    instance_id__in=chunk
+                ).aggregate(total=Sum('media_file_size'))
+                if form_attachments['total']:
+                    form_attachments_total += form_attachments['total']
+
+            if form_attachments_total:
+                with transaction.atomic():
+                    if self.verbosity >= 1:
+                        self.stdout.write(
+                            f'\tUpdating xform attachment storage to '
+                            f"{form_attachments_total} bytes"
+                        )
+
+                    XForm.objects.select_for_update().filter(
+                        pk=xform['pk']
+                    ).update(
+                        attachment_storage_bytes=form_attachments_total
+                    )
```

Comment:
1. What is it that's supposed to succeed or fail as a unit within this transaction block?
2. `update()` is already atomic; I'm not sure what `select_for_update()` contributes.
3. If this is motivated by a desire for locking beyond `submissions_suspended` (perhaps to stop people from deleting while counting is in progress), then the `XForm` needs to be locked before the whole chunking process begins.